### PR TITLE
Types/three 0.149.0 update

### PR DESCRIPTION
**Pull request by epreston**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [three.js 148 - 149 migration guide](https://github.com/mrdoob/three.js/wiki/Migration-Guide)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/three/examples/jsm/helpers/ViewHelper.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import {
+    Camera,
+	Object3D,
```
Comment (DangerBotOSS) [inline types/three/examples/jsm/helpers/ViewHelper.d.ts:3]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## three ([<kbd>unpkg</kbd>](https://unpkg.com/browse/three@latest/))
was missing the following properties:
1. BoxBufferGeometry
1. CapsuleBufferGeometry
1. CircleBufferGeometry
1. ConeBufferGeometry
1. CylinderBufferGeometry

<details>
<summary>as well as these 23 other properties...</summary>
<p>DataUtils, DodecahedronBufferGeometry, ExtrudeBufferGeometry, IcosahedronBufferGeometry, LatheBufferGeometry, LinearSRGBColorSpace, NoColorSpace, OctahedronBufferGeometry, PlaneBufferGeometry, PolyhedronBufferGeometry, RED_GREEN_RGTC2_Format, RED_RGTC1_Format, RingBufferGeometry, SIGNED_RED_GREEN_RGTC2_Format, SIGNED_RED_RGTC1_Format, SRGBColorSpace, ShapeBufferGeometry, SphereBufferGeometry, TetrahedronBufferGeometry, TorusBufferGeometry, TorusKnotBufferGeometry, TubeBufferGeometry, TwoPassDoubleSide</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f85fa3a142570a8f663b3dd4a479b29644c6c549
</p>

File: types/three/examples/jsm/helpers/ViewHelper.d.ts (hunk 1 above)
Comment (epreston) [reply types/three/examples/jsm/helpers/ViewHelper.d.ts]: Updated to spacing.

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Can you open this PR in three-TS-types please?

**Review by joshuaellis — CHANGES_REQUESTED** [review]
.

**Comment by epreston** [comment]
Sure, I can raise it in that repository as well. Feel free to approve this change for this repository.

**Comment by epreston** [comment]
Do we have a list or repositories changes should be replicated to ?

**Comment by joshuaellis** [comment]
> Sure, I can raise it in that repository as well. Feel free to approve this change for this repository.

This is not how we manage the types for this library, the PR should be opened in the original repository and changes then come from that repo to here to ensure we're in sync with a unidirectional flow of information. Thanks :)

**Comment by epreston** [comment]
That's not fair, source material is in the change notes above.  Totally understand redirecting all type changes to another repository.  Let's update the instructions so others don't get confused.

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Changes need to be made at other repo as stated. I don't think these are all the changes for 149 either.

**Comment by epreston** [comment]
They are all the changes for 149.  Problem is, we need to correct most of the existing types and the type tests don't test the types.  

That's a lot of "T"'s to say out loud.  I can do 3 per day, for a few weeks.

That's not true.  You can easily confirm updates with the tracking you have in the repository you pointed me to.  Here:
https://github.com/three-types/three-ts-types/issues/312

**Comment by joshuaellis** [comment]
We don't normally take the migration guide as the lore on what needs updating.

We'd need to revise the code changes via PRs which is what we currently do.

Until I've validated that these are all the changes for the new release I don't think I can approve this PR - so please bare with.

Typically we can do incremental updates for a release in the other repo which is why we prefer to work that way just for some context.

**Comment by epreston** [comment]
If you don't have the ability to check type updates against source, why are you maintaining these ?

Most every type has issues, I'm struggling to see any form of validation process.  Even the tests enforce errors in the types because they use old code not relevant to the version being released.

**Comment by epreston** [comment]
There's a guide linked at the very top that lists each change to public interfaces that affect types.

**Comment by joshuaellis** [comment]
> There's a guide linked at the very top that lists each change to public interfaces that affect types.

I'm going to ignore your other comment because you're clearly frustrated at your experience here.

I see the link and I thank you, what I'm saying is that we need to validate that there are no other changes in the latest release that have not been captured here to ensure we release a high quality update 😊

**Comment by epreston** [comment]
@RyanCavanaugh Maintainers don't action these. This thing is a mess.  Types don't even see the basic example in the projects readme as valid.

**Comment by 0b5vr** [comment]
This has already been addressed @ https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64146

**Pull request closed without merging**